### PR TITLE
Feature/preview image on hover

### PR DESCRIPTION
**Pull request by Hoookie**

### CODE REVIEW

File: src/app/shared/components/image-preview-tooltip/image-preview-tooltip.component.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+})
+export class ImagePreviewTooltipComponent implements AfterViewInit {
+  private _initialStyles: any = {};
+
+  @ViewChild('tooltip', { static: false }) tooltip;
+
+  src: string;
+  set styles(value: { left: string, top: string }) {
+    if (this.tooltip) {
+      this.renderer.setStyle(
```
Comment (viento006) [inline src/app/shared/components/image-preview-tooltip/image-preview-tooltip.component.ts]: can this be moved into tooltip directive?

File: src/app/shared/components/image-preview-tooltip/image-preview-tooltip.component.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      );
+    } else {
+      this._initialStyles = value;
+    }
+  }
+
+  constructor(private renderer: Renderer2) { }
+
+  ngAfterViewInit() {
+    this.styles = this._initialStyles;
```
Comment (viento006) [inline src/app/shared/components/image-preview-tooltip/image-preview-tooltip.component.ts]: bad idea overriding set property with plain value. You can catch somany bugs if you try to extend this component's functionality

File: src/app/shared/directives/image-preview.directive.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  private buildTooltipStyles(event: MouseEvent) {
+    const tooltip = (this.component.instance.tooltip.nativeElement as HTMLElement);
+    const tooltipSizes = tooltip.getClientRects()[0];
+
+    const x = event.x - tooltipSizes.width / 2;
+    const y = event.y + 25;
+
+    return {
+      top: y + 'px',
```
Comment (viento006) [inline src/app/shared/directives/image-preview.directive.ts]: use interpolation over concatenation

File: src/app/shared/directives/image-preview.directive.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    private applicationRef: ApplicationRef,
+    @Inject(DOCUMENT) private document: any
+  ) {}
+
+  private buildTooltipStyles(event: MouseEvent) {
+    const tooltip = (this.component.instance.tooltip.nativeElement as HTMLElement);
+    const tooltipSizes = tooltip.getClientRects()[0];
+
+    const x = event.x - tooltipSizes.width / 2;
+    const y = event.y + 25;
```
Comment (viento006) [inline src/app/shared/directives/image-preview.directive.ts]: magic number

File: src/app/shared/directives/image-preview.directive.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  constructor(
+    private componentFactory: ComponentFactoryResolver,
+    private viewContainer: ViewContainerRef,
+    private applicationRef: ApplicationRef,
+    @Inject(DOCUMENT) private document: any
+  ) {}
+
+  private buildTooltipStyles(event: MouseEvent) {
+    const tooltip = (this.component.instance.tooltip.nativeElement as HTMLElement);
+    const tooltipSizes = tooltip.getClientRects()[0];
```
Comment (viento006) [inline src/app/shared/directives/image-preview.directive.ts]: [ tooltipSizes ] = ...

File: src/app/shared/directives/image-preview.directive.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  @HostListener('mouseover', ['$event']) onHover(event: MouseEvent) {
+    if (this.isShown) {
+      return;
+    }
+    this.showTooltip(event);
+  }
+
+  @HostListener('mousemove', ['$event']) moveImage(event: MouseEvent) {
+    this.component.instance.styles = this.buildTooltipStyles(event);
```
Comment (viento006) [inline src/app/shared/directives/image-preview.directive.ts]: check how often is triggered this handler. possible performance leak

Comment (Hoookie) [reply src/app/shared/directives/image-preview.directive.ts]: Only inside element on which directive was applied

**Pull request merged**